[Farrell Nah] iP

boundtotheearth opened the pull request.

boundtotheearth committed:
Add Level-1 Requirements

Add the features for a skeletal program as specificed in Level-1

Let's
* Make the program greet the user
* Make the program echo commands
* Make the program exit with the 'bye' command

See below for more details.
https://nus-cs2103-ay2021s1.github.io/website/schedule/week2/project.html

boundtotheearth committed:
Commands: Use switch-case instead of if statements

The program checks for commands in user input using
if statements.

Adding more commands will make the code harder to
read.

Let's use switch case statements to check for commands
instead of if statements

boundtotheearth committed:
Add Level-2 Requirements

The program echos user input.

The program should remember user input and display them
back to the user when requested.

Let's
* Add a 'list' command that lists previous commands
* Remember all input that is not 'list' and 'bye'

See below for more information
https://nus-cs2103-ay2021s1.github.io/website/schedule/week2/project.html

boundtotheearth committed:
Use a class to represent tasks

Tasks are represented as strings.

Tasks can contain other properties such as completion status.
Using classes will allow simpler implementation of these
properties.

Let's
* Create a Task class
* Change existing code to use the Task class
* Add an additional 'isDone' property to track the completion

boundtotheearth committed:
Minor changes

* Task list starts at 0
* Entire user input is checked to match a command

* Task list should start at 1 for natural counting
* Future commands may require additional parameters,
  so only the first word should be checked, with the
  rest used as parameters

Let's
* Start counting task list from 1
* Only check first word of user input for matching
  command

boundtotheearth committed:
Add Level-3 Requirements, Bugfixes

Bug: Only first word of task is stored as task description

Let's
* Add a 'done' command
* Add the ability to mark tasks as done
* Fix the above bug

See below for more information
https://nus-cs2103-ay2021s1.github.io/website/schedule/week2/project.html

boundtotheearth committed:
Add todo, event, deadline

Add todo, event and deadline classes in anticipation
for the Level-4 requirements.

Let's add classes for todo, events and deadlines that
inherit from the Task class

Use inheritance to reduce code duplication, such as for
printing tasks and marking as done

boundtotheearth committed:
Add Level-4 Requirements

Let's
* Add support for ToDos
* Add support for Deadlines
* Add support for Events

boundtotheearth committed:
Add Automated Testing

Let's
* Add automated testing using I/O redirection

boundtotheearth committed:
Add Level-5 Requirements

Implement the mimimal requirements for error handling

Let's
* Create a DukeException class to encapsulate error data
* Extract command processing logic to its own function to
  allow for exceptions
* Implement error handling for missing todo descriptions
* Implement error hanlding for unknown commands
* Update tests to check for error handling

See below for more information
https://nus-cs2103-ay2021s1.github.io/website/schedule/week2/project.html

boundtotheearth committed:
Add Level-6 Requirements, bugfixes

Bug: Program does not exit properly after 'bye'
command is given.

Let's
* Add support for deleting tasks from the list
* Update tests to check new features
* Fix the above bug

See below for more information
https://nus-cs2103-ay2021s1.github.io/website/schedule/week2/project.html

boundtotheearth committed:
Commands: Use enums instead of strings

Commands are processed as strings directly from user input

Commands can be handled in a case insensitive way using enums

Let's convert the user input to an enum before processing

boundtotheearth committed:
Add Level-7 Requirements

Task data is not saved to hard disk, so data is
lost when the user quits and restarts the program

A save and load feature allows user data to
persist through program runs.

Let's
* add a DataManager class to manage file io
* save task data when the user updates the
  task list
* load the saved task data when the program
  starts

boundtotheearth committed:
Add Level-8 Requirements

Represent dates as LocalDate instead of String

Let's
* store date data as LocalDate objects instead of string
* format date outputs according to locale
* accept date inputs in the format: "yyyy-mm-dd"

boundtotheearth committed:
Merge branch 'branch-Level-8'

boundtotheearth committed:
Fix date parsing bug

Dates are parsed and formatted as strings instead of LocalDate objects

Let's parse and format dates as LocalDate objects

boundtotheearth committed:
Extract task list features

All task list features are handled in the main program.
(e.g. adding, deleting, updating)

Extracting these features to their own object will make
it easier to manage in the future.

Let's
* Extract task list related features to a TaskList class
* Update save/load to work with the new class

boundtotheearth committed:
Extract UI Features

User I/O is being handled in the main program

Extracting user I/O to its own class will make
it easier to manage

Let's create anew UIManager class to handle
user I/O

boundtotheearth committed:
Organize classes into duke package

boundtotheearth committed:
Add JUnit Tests

Let's
* Add test for displaying startup message
* Add test for printing in window
* Add test for creating todo

boundtotheearth committed:
Setup Gradle

boundtotheearth committed:
Setup Gradle

boundtotheearth committed:
Merge branch 'master' of https://github.com/boundtotheearth/ip

boundtotheearth committed:
Add Level-9 Requirements

Let's give users a way to find a task by
searching for a keyword

boundtotheearth committed:
Cleanup Code

Code style is inconsistent. Some lines
are too long.

Code should follow the established
conventions to make it easier to read.

Lets clean up areas that are not following
the code convention.

boundtotheearth committed:
Add JavaDoc comments

boundtotheearth committed:
Merge branch 'branch-A-JavaDoc'

boundtotheearth committed:
Merge branch 'branch-A-CodingStandard'

boundtotheearth committed:
Merge branch 'branch-Level-9'

boundtotheearth committed:
Add Checkstyle

Let's
* Add checkstyle and suppression files
* Add checkstyle plugin
* Add checkstyle using gradle

boundtotheearth committed:
Fix style violations

boundtotheearth committed:
Add Level-10 Requirements

Let's
* Refactor original duke class to work with GUI
* Add GUI using JavaFX

boundtotheearth committed:
Merge branch 'branch-Level-10'

jzwoo commented on src/main/java/farrell/duke/TaskList.java:
```diff
@@ -0,0 +1,103 @@
+     */
+    public String filteredToString(String matchString) {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < taskList.size(); i++) {
```
Deep nesting here. Avoid more than 3 levels of indentation.

jzwoo commented on src/main/java/farrell/duke/DataManager.java:
```diff
@@ -0,0 +1,116 @@
+            return new TaskList();
+        }
+
+        try {
```
Deep nesting here. Maybe can simplify it a little to increase readability?

jzwoo commented on src/main/java/farrell/duke/DataManager.java:
```diff
@@ -0,0 +1,116 @@
+    public void save(TaskList taskList) throws DukeException {
+        List<Task> tasks = taskList.getAllTasks();
+        try {
+            File dataFile = new File(filePath);
```
Maybe can use a line spacing once in awhile to improve readability?

boundtotheearth committed:
Fix Bugs

* App crashes if data directory does not exist on first run.
* DukeTest is refering to wrong main method

Let's fix the above bugs

boundtotheearth committed:
Fix Bugs

* App crashes if data directory does not exist on first run.
* DukeTest is refering to wrong main method

Let's fix the above bugs

boundtotheearth committed:
Add Assertions

boundtotheearth committed:
Cleanup Code

* Remove unsed code
* Extract functions to reduce indentation and improve
  readibility

boundtotheearth committed:
Code Cleanup

* Added JavaDoc comments
* Fix whitespace issues

boundtotheearth committed:
Merge pull request #2 from boundtotheearth/branch-A-CodeQuality

Cleanup Code

boundtotheearth committed:
Merge branch 'master' into branch-A-Assertions

boundtotheearth committed:
Merge pull request #1 from boundtotheearth/branch-A-Assertions

Branch-A-Assertions

boundtotheearth committed:
Implement Task Sorting

Tasks are sorted in order when they are added.

Users should be able to sort tasks by their
description and time.

Let's
* Create a TimedTask class that event and deadline
  can extend from
* Create a new sort command, with optional parameter
  /order
* Allow users to sort tasks by description and
  time
* Allow users to change the order of the sort

boundtotheearth committed:
Abstracted Command Parsing

Commands are parsed in main program.

Commands should be abstracted so that
adding and editing command behaviour
is easier.

Let's
* Create an abstract Command class
* Create classes for each command
* Change main program to execute
  commands using the new class

boundtotheearth committed:
Add Packages

Let's put classes in relevant packages and refactor code
to work with the new packages.

boundtotheearth committed:
Add More Tests

boundtotheearth committed:
Improve GUI

boundtotheearth committed:
Add More Error Messages

boundtotheearth committed:
Update README

boundtotheearth committed:
Update JavaDoc comments

boundtotheearth committed:
Update README
